statuspage_connector.py

```diff
@@ -0,0 +1,591 @@
+                message = "Error from server. Status Code: {0} Data from server: {1}".format(
+                    r.status_code, resp_json.get('errors', [])[0][0].get('message'))
+        except Exception:
+            pass
+        return RetVal(action_result.set_status(phantom.APP_ERROR, message), None)
+
+    def _process_response(self, r, action_result):
+
+        # store the r_text in debug data, it will get dumped in the logs if the action fails
+        if hasattr(action_result, 'add_debug_data'):
+            action_result.add_debug_data({'r_status_code': r.status_code})
+            action_result.add_debug_data({'r_text': r.text})
+            action_result.add_debug_data({'r_headers': r.headers})
+
+        # Process each 'Content-Type' of response separately
+
+        # Process a json response
+        if 'json' in r.headers.get('Content-Type', ''):
+            return self._process_json_response(r, action_result)
+
+        # Process an HTML resonse, Do this no matter what the api talks.
+        # There is a high chance of a PROXY in between SOAR and the rest of
+        # world, in case of errors, PROXY's return HTML, this function parses
+        # the error and adds it to the action_result.
+        if 'html' in r.headers.get('Content-Type', ''):
+            return self._process_html_response(r, action_result)
+
+        # it's not content-type that is to be parsed, handle an empty response
+        if not r.text:
+            return self._process_empty_reponse(r, action_result)
+
+        # everything else is actually an error at this point
+        message = "Can't process response from server. Status Code: {0} Data from server: {1}".format(
+                r.status_code, r.text.replace('{', '{{').replace('}', '}}'))
+
+        return RetVal(action_result.set_status(phantom.APP_ERROR, message), None)
+
+    def _make_rest_call(self, endpoint, action_result, headers=None, params=None, data=None, json=None, method="get", files=None):
+
+        resp_json = None
+
+        if json:
+            api_key = "API Key {}".format(self._api_key)  # pragma: allowlist secret
+            headers = {'Content-Type': 'application/json', 'Authorization': api_key}
+        else:
+            api_key = "API Key {}".format(self._api_key)  # pragma: allowlist secret
+            headers = {'Authorization': api_key}
+
+        try:
+            request_func = getattr(requests, method)
+        except AttributeError:
+            return RetVal(action_result.set_status(phantom.APP_ERROR, "Invalid method: {0}".format(method)), resp_json)
+
+        # Create a URL to connect to
+        url = "{}{}".format(self._base_url, endpoint)
+
+        try:
+            r = request_func(
+                        url,
+                        data=data,
+                        json=json,
+                        headers=headers,
+                        verify=True,
+                        params=params)
```

Comment:
Shouldn't we add timeout in the REST API call? As per new practice, we are adding 30-sec timeout.

Reply:
Added the timeout parameter to the REST API call, and we used the timeout configuration parameter as a value.